GVFS/GVFS.Common/PlaceholderListDatabase.cs

```diff
@@ -120,11 +126,11 @@ public void RemoveAndFlush(string path)
         ///     - If WriteAllEntriesAndFlush is *not* called entries that were added to the PlaceholderListDatabase after
         ///       calling GetAllEntriesAndPrepToWriteAllEntries will be lost
         /// </remarks>
-        public List<PlaceholderData> GetAllEntriesAndPrepToWriteAllEntries()
+        public List<IPlaceholderData> GetAllEntriesAndPrepToWriteAllEntries()

```

Comment:
Should `GetAllEntriesAndPrepToWriteAllEntries()` be renamed as well (like  `GetAllEntriesAndPrepToWriteAllEntries(out List<PlaceholderData> filePlaceholders, out List<PlaceholderData> folderPlaceholders)` was)?

Reply:
That one I left because it is only a part of the `PlaceholderListDatabase` and not the interface because SQLite didn't necessarily _need_ it since there was already a `GetAllEntries` and  it is only used in tests and in the DiskLayoutUpgrades.  But if you feel differently that it should be part of the interface and renamed let me know.

Reply:
> But if you feel differently that it should be part of the interface

I agree there is no need to add it to the interface (and it would be better not to add it).

> and renamed let me know.

I do think it would be nice to keep the name of this method and the old `GetAllEntriesAndPrepToWriteAllEntries` method in sync as the idea is they are supposed to do the exact same thing (other than returning the data in one collection or two).

I suppose the other option is we delete this method entirely and have everyone use the version that returns separate file and folder placeholders (and the caller can combine them if needed).  Thoughts on that approach?  

Based on a quick glance over how the upgraders are using the placeholder list it doesn't look like it would be too bad to switch them to the method in the interface.

One other question (that is slightly outside the scope of this PR).  Will this:

```
///     - If WriteAllEntriesAndFlush is *not* called entries that were added to the PlaceholderListDatabase after
///       calling GetAllEntriesAndPrepToWriteAllEntries will be lost	        ///       calling GetAllEntriesAndPrepToWriteAllEntries will be lost
```

Still be an issue after moving to SQLite?

Reply:
For this PR I will rename it to keep it in sync and look at deleting in another PR just to reduce the risk and functional changes.

The entries will not be "lost" because they will be added to the database but they will not be in the list of entries returned by `GetAllEntries` so I will need to check those code paths to make sure we don't need those added entries where we called `GetAllEntries`.